planner/core/find_best_task.go

```diff
@@ -1239,6 +1289,27 @@ func (ds *DataSource) convertToPartialTableScan(prop *property.PhysicalProperty,
 		tablePlan.SetChildren(ts)
 		return tablePlan
 	}
+	if matchProp {
+		ts.KeepOrder = true
+		ts.Desc = prop.SortItems[0].Desc
```

Comment:
I think we can pass `isMatchProp = true` to `getOriginalPhysicalTableScan` above to reduce some repeated code.